### PR TITLE
IC3SA: traverse formula iteratively as DAG in justify_coi

### PR DESCRIPTION
**Pull request by makaimann**

### CODE REVIEW

File: engines/ic3sa.cpp [hunk 1]
```diff
@@ -816,43 +816,58 @@ UnorderedTermSet IC3SA::add_to_term_abstraction(const Term & term)
...
+          to_visit.push_back(state_updates.at(curr_c));
+        }
       }
-    }
 
-    UnorderedTermSet free_vars;
-    get_free_symbolic_consts(c, free_vars);
-    for (const auto & fv : free_vars) {
-      projection.insert(fv);
+      UnorderedTermSet free_vars;
```
Comment (ahmed-irfan) [inline engines/ic3sa.cpp:866]: minor: declare this outside of the loop and clear it here

**Review by ahmed-irfan — APPROVED** [review]
LGTM

only one minor comment

File: engines/ic3sa.cpp (hunk 1 above)
Comment (makaimann) [reply engines/ic3sa.cpp]: Good point -- addressed with a1222d000e801ac16210a389e4c769ed5acda662.

Comment (ahmed-irfan) [reply engines/ic3sa.cpp]: even better :)

**Pull request merged**